api/core/rag/datasource/vdb/pgvector/pgvector.py

```diff
@@ -200,7 +204,10 @@ def _create_collection(self, dimension: int):
             with self._get_cursor() as cur:
                 cur.execute("CREATE EXTENSION IF NOT EXISTS vector")
                 cur.execute(SQL_CREATE_TABLE.format(table_name=self.table_name, dimension=dimension))
-                # TODO: create index https://github.com/pgvector/pgvector?tab=readme-ov-file#indexing
+                # DONE: create index https://github.com/pgvector/pgvector?tab=readme-ov-file#indexing
+                # PG hnsw index only support 2000 dimension or less
+                if dimension <= 2000:
+                    cur.execute(SQL_CREATE_INDEX.format(table_name=self.table_name))
```

Comment:
Consider use `sqlalchemy`'s `text` aka `sql_text` method to wrap and templating instead. Or simply format strings.